app/models/purchase.py

```diff
@@ -2,33 +2,77 @@
 from datetime import datetime
 import pytz
 
+
+class CartItem(db.Model):
+    ''' Functions as association table between listings and merchants '''
+    __tablename__ = "cartItems"
+    merchant_id = db.Column(db.Integer, db.ForeignKey('users.id'),
+                            primary_key=True)
+    listing_id = db.Column(db.Integer, db.ForeignKey('listings.id'),
+                           primary_key=True)
+    quantity = db.Column(db.Integer)
+
+    listing = db.relationship("Listing")
+
+
+class Status:
+    PENDING = 0
+    APPROVED = 1
+    DECLINED = 2
+
+
+class Order(db.Model):
+    __tablename__ = 'orders'
+    id = db.Column(db.Integer, primary_key=True)
+
+    date = db.Column(db.DateTime)
+    status = db.Column(db.Integer)
+
+    def __init__(self, cart_items):
+        self.date = datetime.now(pytz.timezone('US/Eastern'))
+        self.status = Status.PENDING
+
+        for item in cart_items:
+            vendor_id = item.listing.vendor
+            listing_id = item.listing.id
+            qty = item.quantity
+            item = item.listing.name
+            cost = item.listing.cost * qty
+            db.session.add(Purchase(vendor_id, listing_id, self, qty, item,
+                                    cost))
+
+        db.session.commit()
+
+    def __repr__(self):
+        return "<Order: {}>".format(self.id)
+
+
 class Purchase(db.Model):
     __tablename__ = 'purchases'
     id = db.Column(db.Integer, primary_key=True)
 
     # model relationships
-    merchant_id = db.Column(db.Integer, db.ForeignKey('users.id'))
+    vendor_id = db.Column(db.Integer, db.ForeignKey('users.id'))
+    order_id = db.Column(db.Integer, db.ForeignKey('orders.id'))
+    order = db.relationship("Order", backref="purchases")
```

Comment:
This relationship, even if it looks strange (b/c nothing coded in `Order`) *should* make it so the `purchases` field in an `Order` is updated automagically when the Purchase is created, provided I understand the docs.